v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -726,9 +726,12 @@ type Subtest struct {
 func (tc *KubePerTestContext) RunSubtests(tests ...Subtest) {
 	for _, test := range tests {
 		test := test
-		tc.T.Run(test.Name, func(t *testing.T) {
+		passed := tc.T.Run(test.Name, func(t *testing.T) {
 			test.Test(tc.Subtest(t))
 		})
+		if !passed {
+			tc.T.Fatalf("subtest %s failed", test.Name)
+		}
```

Comment:
I realized that by default, go test doesn't fail the parent test if the child test fails.

Until now we haven't really cared because we usually run the child tests at or near the end of the parent test anyway and so we haven't noticed that test execution actually proceeds after a child test fails. In this case I had a test like this:

```
// Main test

// run subtest in sequence

// run N subtests in parallel
```

and the subtest in sequence part failed and gobbled some recordings that the parallel tests needed, causing them to spin forever.

This moves the subtest pattern to match gomega.Expect, which terminates the test right then and there and is I think what we actually expect.

Probably I need to fix the parallel subtests helper to do this too...

Reply:
aborting at the first failure helps debugability too, because it avoids situations like:

<real error>
... 
<100's of lines of errors from bits that ran after the real error but are really because the state of the resource for this assertion was not valid, due to "real error" above>

which makes it hard to find the real cause of the problem.

Reply:
Ah, that makes sense. I thought maybe it was debris from debugging.